lib/rules/utils/ast-utils.js

```diff
@@ -1006,6 +1006,42 @@ function isTopLevelExpressionStatement(node) {
 
 }
 
+/**
+ * Check whether the given node is a directive or not.
+ * @param {ASTNode} node The node to check.
+ * @param {SourceCode} sourceCode The source code object to get tokens.
+ * @returns {boolean} `true` if the node is a directive.
+ */
+function isUnparenthesizedStringLiteral(node, sourceCode) {
+    return node.type === "Literal" && typeof node.value === "string" && !isParenthesised(sourceCode, node);
+}
+
+/**
+ * Check whether the given node is a part of a directive prologue or not.
+ * @param {ASTNode} node The node to check.
+ * @param {SourceCode} sourceCode The source code object to get tokens.
+ * @returns {boolean} `true` if the node is a part of directive prologue.
+ */
+function isDirective(node, sourceCode) {
+    if ("directive" in node) {
+        return true;
+    }
+
+    // Fallback for parsers that don't set a `directive` property.
```

Comment:
Not sure if the following fallback logic is necessary. It seems that all up-to-date parsers for ESLint do set a `directive` property on directives. If that's the case then this fallback would be only relevant to obsolete parsers such as Esprima 3.0 and older custom parsers.

Reply:
The `directive` property was added to the ESTree spec in 2017 (https://github.com/estree/estree/commit/3d0f13b783205998485779418f09ecc0fa006126). I think we can assume that parsers now support this property and try this without the fallback. To start with just the two rules that this PR targets, to see if there is any negative feedback, and then we could update other rules if there were no problems reported.

@nzakas what do you think?

Reply:
I'm comfortable with relying on the `directive` property. :+1:

Reply:
@fasttime it seems that we all agree that the fallback can be removed.

Reply:
Thanks for the feedback. I removed the fallback logic.